node/actors/internal_nats_server.go

```diff
@@ -25,23 +45,64 @@ type agentCredential struct {
 	nkey         nkeys.KeyPair
 }
 
+type configTemplateData struct {
+	Credentials       map[string]*credentials
+	Connections       map[string]*nats.Conn
+	NexHostUserPublic string
+	NexHostUserSeed   string
+}
+
+type credentials struct {
+	WorkloadType string
+	NkeySeed     string
+	NkeyPublic   string
+}
+
 func (ns *internalNatsServer) Init(args ...any) error {
 	ns.tokens = make(map[gen.Atom]gen.Ref)
 
-	ns.Log().Info("Internal NATS server started")
+	if len(args) != 1 {
+		err := errors.New("node options are required")
+		ns.Log().Error("Failed to start internal NATS server", slog.String("error", err.Error()))
+		return err
+	}
+
+	if _, ok := args[0].(models.NodeOptions); !ok {
+		err := errors.New("args[0] must be valid node options")
+		ns.Log().Error("Failed to start internal NATS server", slog.String("error", err.Error()))
+		return err
+	}
 
 	ns.nodeOptions = args[0].(models.NodeOptions)
 
+	hostUser, err := nkeys.CreateUser()
+	if err != nil {
+		return err
+	}
+	ns.hostUser = hostUser
+
 	creds, err := ns.buildAgentCredentials()
 	if err != nil {
 		return err
 	}
 	ns.creds = creds
-	err = ns.startNatsServer(creds)
+
+	opts, err := ns.generateConfig()
+	if err != nil {
+		return err
+	}
+
+	err = ns.startNatsServer(opts)
 	if err != nil {
 		return err
 	}
-	_ = ns.Send(ns.PID(), "post_init")
+
+	err = ns.Send(ns.PID(), "post_init")
```

Comment:
we should probably adopt the pattern from the other PR and remove this magic string and replace with a constant `PostInit`

Reply:
See #389.